### PR TITLE
feat!: add Angular 17 support

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2700498](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2700498)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1949/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1949/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1949/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1949/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1949/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1949/playground/)

File: libs/components/storybook/src/lib/preview-wrapper/preview-wrapper-decorators.ts [hunk 1]
```diff
@@ -18,7 +19,7 @@ export const previewWrapperDecorators: DecoratorFunction<
   unknown[]
 >[] = [
   moduleMetadata({
-    imports: [CommonModule, PreviewWrapperModule],
+    imports: [CommonModule, PreviewWrapperModule, RouterTestingModule],
```
Comment (Blackbaud-SteveBrush) [inline libs/components/storybook/src/lib/preview-wrapper/preview-wrapper-decorators.ts:22]: Is there something different in Angular 17 that needed this module imported?

File: libs/components/validation/src/lib/modules/email-validation/fixtures/email-validation.component.fixture.html [hunk 2]
```diff
@@ -8,7 +8,7 @@
     *ngIf="email.errors && email.errors.skyEmail && (email.dirty || email.touched)"
   >
     <div [hidden]="!email.errors.skyEmail.invalid">
-      Please enter valid email with format joe@abc.com
+      Please enter valid email with format joe&#64;abc.com
```
Comment (Blackbaud-SteveBrush) [inline libs/components/validation/src/lib/modules/email-validation/fixtures/email-validation.component.fixture.html:11]: I'm seeing a few of these; what's the background on the change?

File: package.json [hunk 3]
```diff
@@ -153,68 +153,74 @@
...
-    "ts-jest": "29.1.1",
-    "ts-node": "10.9.1",
-    "typescript": "5.1.6"
+    "storybook": "^7.6.10",
+    "ts-jest": "29.1.2",
+    "ts-node": "10.9.2",
+    "typescript": "5.3.3"
+  },
+  "overrides": {
+    "ng2-dragula@5.0.1": {
```
Comment (Blackbaud-SteveBrush) [inline package.json:220]: What's the reason for this change?

File: libs/components/storybook/src/lib/preview-wrapper/preview-wrapper-decorators.ts (hunk 1 above)
Comment (johnhwhite) [reply libs/components/storybook/src/lib/preview-wrapper/preview-wrapper-decorators.ts:22]: Doing this allowed me to drop the dependency on `storybook-addon-angular-router`, which didn't work with the updated Storybook.

File: libs/components/validation/src/lib/modules/email-validation/fixtures/email-validation.component.fixture.html (hunk 2 above)
Comment (johnhwhite) [reply libs/components/validation/src/lib/modules/email-validation/fixtures/email-validation.component.fixture.html:11]: The new template syntax uses `@` as a control character, so Angular has a schematic to escape them. I believe this change is optional.

File: package.json (hunk 3 above)
Comment (johnhwhite) [reply package.json:220]: `ng2-dragula` does not yet have Angular 17 support. The [Angular 17 support PR](https://github.com/valor-software/ng2-dragula/pull/1101) is currently in progress. This is a workaround.

File: libs/components/storybook/src/lib/preview-wrapper/preview-wrapper-decorators.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/storybook/src/lib/preview-wrapper/preview-wrapper-decorators.ts:22]: Cool.

**Review by Blackbaud-SteveBrush — APPROVED** [review]
